### ISSUE TITLE
'Load' type is missing the 'cookies' object

### ISSUE THREAD

**Issue by JosephBrooksbank** [sveltejs/kit#8185]
### Describe the bug

The following code works in my base `+page.server.ts` file:
```javascript
export const load = async ({cookies}) => {
     console.log(cookies.get('cookieName');
}
```
but obviously I get type warnings and issues. If I change it to 
```typescript
import type {Load} from '@sveltejs/kit';

export const load: Load = async ({cookies}) => {
	console.log(cookies.get('cookieName'));
}
```

The code still works, but I have a constant error of :
```
Property 'cookies' does not exist on type 'LoadEvent<Partial<Record<string, string>>, Record<string, any> | null, Record<string, any>, string | null>'.
```
Which does not appear when I use other data, like `fetch` or `setHeaders`.

### Reproduction

https://stackblitz.com/edit/sveltejs-kit-template-default-vqdsm2?file=src%2Froutes%2F%2Bpage.server.ts&terminal=dev You can see the cookie is stored properly by refreshing the preview and checking the console

### System Info

```Shell
System:
    OS: Linux 5.0 undefined
    CPU: (8) x64 Intel(R) Core(TM) i9-9880H CPU @ 2.30GHz
    Memory: 0 Bytes / 0 Bytes
    Shell: 1.0 - /bin/jsh
  Binaries:
    Node: 16.14.2 - /usr/local/bin/node
    Yarn: 1.22.19 - /usr/local/bin/yarn
    npm: 7.17.0 - /usr/local/bin/npm
  npmPackages:
    @sveltejs/adapter-auto: ^1.0.0 => 1.0.0 
    @sveltejs/kit: ^1.0.0 => 1.0.0 
    svelte: ^3.54.0 => 3.55.0 
    vite: ^4.0.0 => 4.0.1 
```
```


### Severity

annoyance

### Additional Information

_No response_

**Comment by Rich-Harris** [comment]
This is expected. [`Load`](https://kit.svelte.dev/docs/types#public-types-load) is the generic form of `PageLoad` and `LayoutLoad`, which is what you should use in `+page.js` and `+layout.js` files. In these files, you don't have access to `cookies` because the `load` function can run in the browser, where cookies cannot be accessed by JavaScript for security reasons (unless you set `httpOnly: false`, which you really shouldn't, but that's a different story).

In `+page.server.js` and `+layout.server.js`, which never run in the browser, you should use `PageServerLoad` and `LayoutServerLoad` which extend [`ServerLoad`](https://kit.svelte.dev/docs/types#public-types-serverload).

In short: use the generated types in `./$types`, not the generic ones from `@sveltejs/kit`. You'll find that it won't even let you import the wrong thing — it will only make `PageServerLoad` and `LayoutServerLoad` available in the first place, so you'll never run into this.

**Comment by JosephBrooksbank** [comment]
Thanks! That makes sense. I couldn't really find much direct documentation on `load`, so I was working off old github issues. Those must have been for `+page.js`

**Comment by NielAgi** [comment]
I don't get this.
I open a new page.server.ts file and enter this code:
```
import type { PageServerLoad } from './$types';

export const load = (async () => {
    console.log(cookies.get('user'));
    return {};
}) satisfies PageServerLoad;
```

I get the error: " Cannot find name 'cookies'".
What am I missing?

**Comment by brunnerh** [comment]
You did not declare the event or destructure the relevant property?

```diff
- export const load = (async () => {
+ export const load = (async ({ cookies }) => {
```
